mllib/src/main/scala/org/apache/spark/ml/feature/CountVectorizer.scala

```diff
@@ -152,16 +152,10 @@ class CountVectorizer(override val uid: String)
       (word, count)
     }.cache()
     val fullVocabSize = wordCounts.count()
-    val vocab: Array[String] = {
-      val tmpSortedWC: Array[(String, Long)] = if (fullVocabSize <= vocSize) {
-        // Use all terms
-        wordCounts.collect().sortBy(-_._2)
-      } else {
-        // Sort terms to select vocab
-        wordCounts.sortBy(_._2, ascending = false).take(vocSize)
-      }
-      tmpSortedWC.map(_._1)
-    }
+
+    val vocab: Array[String] = wordCounts.
+    top(math.min(fullVocabSize, vocSize).toInt)(Ordering.by(_._2))
```

Comment:
@lionelfeng Please use the following style (moving `.` to next line and fix indentation):

~~~scala
val vocab: Array[String] = wordCounts
  .top(math.min(fullVocabSize, vocSize).toInt)(Ordering.by(_._2))
  .map(_._1)
~~~